### ISSUE TITLE
No such file or directory target/classes/log4j.xml

### ISSUE THREAD

**Issue by cgerstle-droid** [cobbzilla/s3s3mirror#108]
Been using s3s3mirror for years and love it. Some time between 09-Oct-2020 and today (04-Jan-2021) something broke/changed in that now, out of the box, I get the following error:
target/classes/log4j.xml (No such file or directory)

Eg:
./s3s3mirror.sh -v s3source s3destination
log4j:ERROR Could not parse url [file:target/classes/log4j.xml].
java.io.FileNotFoundException: target/classes/log4j.xml (No such file or directory)

Using the latest code (16-Dec-2020). I was able to hack around this by modifying s3s3mirror.sh to swap
java ${VERSION_ARG} -Dlog4j.configuration=file:target/classes/log4j.xml -jar "${JARFILE}" "$@"
for 
java ${VERSION_ARG} -Dlog4j.configuration=file:${THISDIR}/./src/main/resources/log4j.xml -jar "${JARFILE}" "$@"

in short, pointing directly at the actual location of log4j.xml. This was directly after performing a git clone to reaquire the codebase. My guess is that I should not have had to perform this hack - true?

**Comment by cobbzilla** [comment]
@cgerstle-droid Yes, most likely a recent PR broke it. I'll look into it.

**Comment by cobbzilla** [comment]
@cgerstle-droid I updated the dependencies, rebuilt the jar and committed it -- can you confirm that this problem is resolved for you?

**Comment by cgerstle-droid** [comment]
Did a fresh git clone, as shown with:
git status# On branch masternothing to commit, working directory clean
git logcommit 679374e2eee8ca5a7d080d9dc6346ca274413d09Author: Jonathan Cobb <jonathan@kyuss.org>Date:   Wed Jan 6 08:23:40 2021 -0500
    update copyright year
commit d8561c52937f4856865b0ccd312260f69fb5240dAuthor: Jonathan Cobb <jonathan@kyuss.org>Date:   Wed Jan 6 08:14:46 2021 -0500
    update jar for 1.2.7
commit cc848341834d86e0ef70abca698b4e3cd8f18688Author: Jonathan Cobb <jonathan@kyuss.org>Date:   Wed Jan 6 08:03:51 2021 -0500
    update link to 2.x version
commit 477b660e576e22adc2f98316682e067b1ef94647Author: Jonathan Cobb <jonathan@kyuss.org>Date:   Wed Jan 6 08:02:51 2021 -0500
    require java8, upgrade dependencies, re-include log4j.xml in jar
commit eb23d65229c3cf334da75ed125b5885176d46bfaAuthor: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>Date:   Wed Dec 16 17:50:49 2020 -0500
    Bump junit from 4.8 to 4.13.1 (#107)
    Bumps [junit](https://github.com/junit-team/junit4) from 4.8 to 4.13.1.    - [Release notes](https://github.com/junit-team/junit4/releases)    - [Changelog](https://github.com/junit-team/junit4/blob/main/doc/ReleaseNotes4.13.1.md)    - [Commits](https://github.com/junit-team/junit4/compare/r4.8...r4.13.1)
    Signed-off-by: dependabot[bot] <support@github.com>
    Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>


But when I tried using the code:
./s3s3mirror.sh -v bucket-1 bucket-2

Exception in thread "main" java.lang.UnsupportedClassVersionError: org/cobbzilla/s3s3mirror/MirrorMain : Unsupported major.minor version 52.0        at java.lang.ClassLoader.defineClass1(Native Method)        at java.lang.ClassLoader.defineClass(ClassLoader.java:808)        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)        at java.net.URLClassLoader.defineClass(URLClassLoader.java:443)        at java.net.URLClassLoader.access$100(URLClassLoader.java:65)        at java.net.URLClassLoader$1.run(URLClassLoader.java:355)        at java.net.URLClassLoader$1.run(URLClassLoader.java:349)        at java.security.AccessController.doPrivileged(Native Method)        at java.net.URLClassLoader.findClass(URLClassLoader.java:348)        at java.lang.ClassLoader.loadClass(ClassLoader.java:430)        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:323)        at java.lang.ClassLoader.loadClass(ClassLoader.java:363)        at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482)
Arg!
Thanks in advance,Carl
    On Wednesday, January 6, 2021, 08:16:07 AM EST, Jonathan Cobb <notifications@github.com> wrote:  
 
 


@cgerstle-droid I updated the dependencies, rebuilt the jar and committed it -- can you confirm that this problem is resolved for you?

**Comment by cobbzilla** [comment]
@cgerstle-droid That error is due to your Java version. The most recent changes have made s3s3mirror require Java 8 as the minimum version (internally, that is the `52.0` you see in the error message).

You may be running Java 7, which is no longer supported. Try again using Java 8 or higher.

**Comment by diasjorge** [comment]
Hi, I tried with the latest release from git and I'm seeing this error:
```
log4j:ERROR Could not parse url [file:target/classes/log4j.xml].
```
I am running with java 8
```
java -version
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-8u252-b09-1~18.04-b09)
OpenJDK 64-Bit Server VM (build 25.252-b09, mixed mode)
```

**Comment by fernandomatsuosantos** [comment]
> Hi, I tried with the latest release from git and I'm seeing this error:
> 
> ```
> log4j:ERROR Could not parse url [file:target/classes/log4j.xml].
> ```
> 
> I am running with java 8
> 
> ```
> java -version
> openjdk version "1.8.0_252"
> OpenJDK Runtime Environment (build 1.8.0_252-8u252-b09-1~18.04-b09)
> OpenJDK 64-Bit Server VM (build 25.252-b09, mixed mode)
> ```

The issue has being closed but we still having the same issue on current version v2.1.5

**Comment by ChrisjanWust** [comment]
Same here. Issue still breaking with java 11 and 18 on version 2.1.5
